### PR TITLE
frontend(torch): add argsort method

### LINKED ISSUE #9267: argsort

### PR DESCRIPTION
**Pull request by y1450**
closes #9267.
Tracking TODO issue #3612.
References : https://pytorch.org/docs/stable/generated/torch.Tensor.argsort.html#torch.Tensor.argsort

### CODE REVIEW

**Comment by y1450** [comment]
hi @ahmedo42, could you please review my pr. I am afraid otherwise it would become stale.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -3365,6 +3365,55 @@ def test_torch_instance_argmin(
     )
 
 
+# argsort
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="argsort",
+    dtype_value=helpers.dtype_and_values(
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3373]: You can use the same helper function to generate both arrays and axis `helpers.dtype_value_axis` to avoid unnecessary data generation. 
And as `complex` dtypes are not supported by `argsort` you will change `valid` to `numeric`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hello @y1450 
Thanks for contributing to ivy😊

All looks great just a few changes are needed🙂
I'll then quickly review the changes and merge them once solved

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 1 above)
Comment (y1450) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3373]: thanks  for the suggestion, I'll update and mark it.
ps `helpers.dtype_value_axis` didn't return any result in docs, apparently it `helpers.dtype_values_axis` missing a s in 'value'. Although I am surprised how neither sphinx nor github search return any meaningful results and I had grep it locally.

**Comment by y1450** [comment]
I re synced my branch and removed the conflict. To keep the history clean, I merge the new changes in single commit.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -3436,6 +3436,56 @@ def test_torch_instance_argmin(
...
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="argsort",
+    dtype_input_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        max_num_dims=3,
+        min_dim_size=1,
+        max_dim_size=3,
+        min_value=1,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3449]: There is no need to specify the values range.
Also, you can increase the number of `max_num_dims` to a large value assuming `5` as an example. Also, the axis value should be between `-1` and `0` which means you can specify `min_axis` and `max_axis`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey these few changes and the failing tests will pass

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 2 above)
Comment (y1450) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: still failing a test case, I'll try to resolve it locally and update you.

**Comment by y1450** [comment]
did look up on the failing test case, as per my understanding 
dtype_input_axis=(['uint8'], [array(0, dtype=uint8)], None), it is due None being passed dim. I tested the assumption
locally.
image
Any suggestion how can it be resolved, although not sure about how or why dim =None is generated.

**Comment by hirwa-nshuti** [comment]
> did look up on the failing test case, as per my understanding dtype_input_axis=(['uint8'], [array(0, dtype=uint8)], None), it is due None being passed dim. I tested the assumption locally. image Any suggestion how can it be resolved, although not sure about how or why dim =None is generated.
> 
> In the codebase, I could not find a function that test the `dtype_values_axis`.

Remove these three arguments from test data generation it will hopefully work
```
valid_axis=True,
force_int_axis=True,
allow_neg_axes=True,
````

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -3436,6 +3436,53 @@ def test_torch_instance_argmin(
     )
 
 
+# argsort
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="argsort",
+    dtype_input_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3445]:
```suggestion
        available_dtypes=helpers.get_dtypes("numeric"),
        min_num_dims=1,
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Generated arrays should not be of 0-D

**Review by hirwa-nshuti — APPROVED** [review]
Looks good now😊
Thank  you for contributing

**Comment by y1450** [comment]
Thanks for the feedback. Also the comment on the dimensions would be helpful for future implementation.
